pkg/storageincentives/agent.go

```diff
@@ -254,35 +248,43 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 		}
 
 		// write the current phase only once
-
 		if currentPhase != prevPhase {
 
 			a.metrics.CurrentPhase.Set(float64(currentPhase))
 
 			a.logger.Info("entering phase", "phase", currentPhase.String(), "round", round, "block", block)
 
+			a.nodeState.SetPhase(currentPhase)
+			a.nodeState.SetRound(round)
+			a.nodeState.SetBlock(block)
+			a.nodeState.SaveStatus()
+
 			phaseEvents.Publish(currentPhase)
 			if currentPhase == claim {
 				phaseEvents.Publish(sample) // trigger sample along side the claim phase
 			}
 		}
 
 		prevPhase = currentPhase
-
 		mtx.Unlock()
+
 	}
 }
 
 func (a *Agent) reveal(ctx context.Context, storageRadius uint8, sample, obfuscationKey []byte) error {
 	a.metrics.RevealPhase.Inc()
 	err := a.contract.Reveal(ctx, storageRadius, sample, obfuscationKey)
+	a.nodeState.SetFee(a.contract.Fee())
```

Comment:
we could pass the contract as a dependency to the nodeState, so you don't have to call `a.contract.Fee()` here and SetFee can call it internally